x-pack/plugins/rule_registry/server/utils/get_common_alert_fields.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Values } from '@kbn/utility-types';
+import { AlertExecutorOptions } from '../../../alerting/server';
+import { ParsedTechnicalFields } from '../../common/parse_technical_fields';
+import {
+  ALERT_ID,
+  ALERT_UUID,
+  ALERT_RULE_CATEGORY,
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_NAME,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  SPACE_IDS,
+  TAGS,
+  TIMESTAMP,
+} from '../../common/technical_rule_data_field_names';
+
+const commonRuleFieldNames = [
+  ALERT_RULE_CATEGORY,
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_NAME,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  SPACE_IDS,
+  TAGS,
+  TIMESTAMP,
+];
+export type CommonRuleFieldName = Values<typeof commonRuleFieldNames>;
+
+const commonAlertFieldNames = [ALERT_ID, ALERT_UUID];
+export type CommonAlertFieldName = Values<typeof commonAlertFieldNames>;
+
+export type CommonRuleFields = Pick<ParsedTechnicalFields, CommonRuleFieldName>;
+
+export const getCommonRuleFields = (
+  options: AlertExecutorOptions<any, any, any, any, any>
+): CommonRuleFields => {
+  return {
+    [ALERT_RULE_CATEGORY]: options.rule.ruleTypeName,
+    [ALERT_RULE_CONSUMER]: options.rule.consumer,
+    [ALERT_RULE_NAME]: options.rule.name,
+    [ALERT_RULE_PRODUCER]: options.rule.producer,
+    [ALERT_RULE_TYPE_ID]: options.rule.ruleTypeId,
+    [ALERT_RULE_UUID]: options.alertId,
```

Comment:
We'll probably want to change this to `ALERT_RULE_ID` and remove `ALERT_RULE_UUID` entirely to align with the [schema](https://docs.google.com/spreadsheets/d/1Cqv0bRx1BgY6VP9PY4FK6kppUEZJ5qMPoFgr5_XQXJI/edit#gid=1775019155)

Reply:
Ok, I'll check if it's used anywhere. (probably not)

Reply:
Per slack conversation, let's stick with `ALERT_RULE_UUID` here and plan to remove `ALERT_RULE_ID` elsewhere.

Reply:
awesome, thanks for following up